src/utils/inference/markupInference.ts

```
@@ -540,21 +560,23 @@ function containerChildren(
   $container: JQuery,
   selected: HTMLElement[],
 ): HTMLElement[] {
-  return selected.map((element) => {
-    const exactMatch = $container.children().filter(function () {
-      return this === element;
-    });
+  return compact(
+    selected.map((element) => {
```

It looks like the logic here was designed to throw the error in the case where a match is not found, rather than return an empty array.

`exactMatch.get(0);` -- can be asserted due to length check, or code could change to pull this and return if not null

`match.get(0);` -- also has a length check, could also be rewritten to return if not null otherwise throw the error



Point of this comment is that I don't want to close one source of bugs while opening another, we may as well keep the logic as "tight" as possible compared to previous.

---

I suppose the `compact()` call just fixes the type and the logic would remain the same though... 🤔 

Oh well, whichever, this works for now I guess, just feels gross to pull in `compact()` only to fix the typing for us 🤷‍♂️

---

Yeah, couldn't get the typing to work otherwise. Just another annoying limitation with arrays